ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -2351,7 +2375,16 @@ inline core::smart_refctd_ptr<asset::ICPUDescriptorSet> CMitsubaLoader::createDS
 		uint32_t instDataOffset = instanceData.size();
 		for (const auto& inst : meta->getInstances()) {
 			emissive = inst.emitter.type==CElementEmitter::AREA ? inst.emitter.area.radiance : core::vectorSIMDf(0.f);
-			instanceData.push_back({inst.tform, inst.instrOffsetCount, core::rgb32f_to_rgb19e7(emissive.pointer)});
+
+			SInstanceData instData;
+			instData.tform = inst.tform;
+			instData.tform.getSub3x3InverseTranspose(instData.normalMatrix);
+			instData.emissive = core::rgb32f_to_rgb19e7(emissive.pointer);
+			//ehh this might get broken by -ffast-math
```

Comment:
why? what's wrong with the implementation of `core::floatBitsToUint` ?

Reply:
not with `floatBitsToUint` but as you were saying once it's better to keep all as uints because `-ffast-math` might change some bits while copying or something (if the bits happen to match ieee754 NaN or inf bit pattern)

Reply:
AFAIK when preforming assignment/copy everything SHOULD be fine, its just performing any other operation with them might backfire